File: javascript/prime-factors/prime-factors.js
//
// This is only a SKELETON file for the 'Prime Factors' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const primeFactors = (number) => {
  let factors = [];
  let factor = 2;
  while (number > 1) {
    if (number % factor === 0) {
      factors.push(factor);
      number /= factor;
    } else {
      factor++;
    };
  };
  return factors;
};

File: cpp/prime-factors/prime_factors.cpp
#include "prime_factors.h"
#include <vector>

namespace prime_factors {
    std::vector<int> of(int num) {
        std::vector<int> factors {};
        int factor{2};
        while (num > 1) {
            if (num % factor == 0) {
                factors.emplace_back(factor);
                num /= factor;
            } else factor++;
        }
        return factors;
    }
}  // namespace prime_factors
